<launch>
   <arg name="multi_robot"             default="robot_0"/>
   <arg name="path_name"               default="sin"/>
   <arg name="follower"                default="PurePursuit"/>
   <arg name="path_csvfile"            default="$(find controller_tutorial)/path/$(arg path_name).csv"/>
   <arg name="repeat_mode"             default="reverse"/>
   <arg name="publish_command"         default="true"/>
   <arg name="use_rviz"                default="true"/>
   <arg name="frame_id_global"         default="map"/>
   <arg name="frame_id_odom"           default="$(arg multi_robot)/odom"/>
   <arg name="topic_cmd"               default="cmd_vel"/>
   <arg name="topic_odom"              default="odom"/>
   <arg name="topic_goal"              default="goal"/>
   <arg name="file_marker"             default="$(find controller_tutorial)/config/target_markers.yaml"/>
   <arg name="max_linear_velocity"     default="0.3"/>
   <arg name="min_linear_velocity"     default="-0.3"/>
   <arg name="max_angular_velocity"    default="1.5"/>
   <arg name="min_angular_velocity"    default="-1.5"/>

   <group ns="$(arg multi_robot)">

      <node pkg="tf" type="static_transform_publisher" name="map_tf_controller" args="0 0 0 0 0 0 $(arg frame_id_global) $(arg frame_id_odom) 60" />

      <node pkg="controller_tutorial"  name="follow_marker" type="follow_marker" output="screen">
         <param name="controller_name" value="potbot_nav/$(arg follower)"/>

         <param name="path_csvfile"    value="$(arg path_csvfile)"/>
         <param name="repeat_mode"     value="$(arg repeat_mode)"/>
         <param name="publish_command" value="$(arg publish_command)"/>
         
         <rosparam file="$(arg file_marker)"  command="load" ns="target" />
         <!-- <param name="target/target_marker/trajectory_recording"    value="true"/> -->

         <param name="controller/max_linear_velocity" value="$(arg max_linear_velocity)"/>
         <param name="controller/min_linear_velocity" value="$(arg min_linear_velocity)"/>
         <param name="controller/max_angular_velocity" value="$(arg max_angular_velocity)"/>
         <param name="controller/min_angular_velocity" value="$(arg min_angular_velocity)"/>

         <remap from="odom"      to="$(arg topic_odom)"/>
         <remap from="cmd_vel"   to="$(arg topic_cmd)"/>
      </node>
      
      <group if="$(arg use_rviz)">
         <node pkg="rviz" type="rviz" name="rviz" required="true" args="-d $(find controller_tutorial)/rviz/follow_marker.rviz"/>
      </group>
   </group>
</launch>
